### DISCUSSION TITLE
Switching between flows

### DISCUSSION THREAD

**Discussion by manikyabard** [NVIDIA-NeMo/Guardrails#271] [Q&A] [answered]
I understand that if we are currently inside a flow (flow A), and if the next user message matches the intent of another flow (flow B), flow B starts getting "executed". After this, if the user's response matches the next user intent in flow A, would it continue from the same point? Is it possible to finish flow A and then go back to flow B?

I saw that the conversation history is part of the intent prompts so this can happen potentially, but is this tested? Is relying on this the only way?

Also, there might be some cases where we don't actually want to "remember" information (variables) in a flow when we start it again. Is there a straightforward method of removing some part of history? (I guess one potential solution is to either redefine those variables, or maybe even prompt to ignore previous conversation)

**Comment by manikyabard** [comment]
@drazvan I looked into the codebase a bit and came across interruptable, extension, response, recursive etc. flows. Are there any examples of using these? I'm assuming that something like 
```
define non-interruptable flow
```
would work, but what would this entail exactly? Some documentation around these concepts or an explanation would be really helpful!

**Reply by drazvan** [reply]
These were experimental "internal" features and will be deprecated very soon. And yes, that's how you would use them.

**Comment by drazvan** [comment, marked as answer]
Regarding the dynamic you mentioned in your first message, yes, flow A gets interrupted and then flow B starts. If at some point flow B finishes, or can't continue, then it is aborted and flow A will, potentially, continue.

**Reply by manikyabard** [reply]
Thanks! Going through some of the next state computation helped make sense of how this happens.
